internal/job/executor.go

```diff
@@ -1033,7 +1026,11 @@ func (e *Executor) checkoutPlugin(ctx context.Context, p *plugin.Plugin) (*plugi
 		return nil, err
 	}
 	// Switch back to the previous working directory
-	defer e.shell.Chdir(previousWd)
+	defer func() {
+		if err := e.shell.Chdir(previousWd); err != nil && e.Debug {
+			e.shell.Errorf("failed to switch back to previous working directory: %v", err)
+		}
```

Comment:
We probably should return the error here. If the working directory is wrong, later plugin checkouts might work. But if this is the last plugin, then we could end up in a difficult to debug state when the job runs.

But, since we have not returned the error for so long, I'm just going to log it.